AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Subscribe/Support/SortedList.swift

```diff
@@ -9,42 +9,52 @@ import Amplify
 
 class SortedList<ModelType: Model> {
     var sortedModels: [ModelType]
+    var modelIds: Set<Model.Identifier>
     private let sortInput: [QuerySortDescriptor]?
     private let modelSchema: ModelSchema
 
     init(sortInput: [QuerySortDescriptor]?, modelSchema: ModelSchema) {
         self.sortedModels = []
+        self.modelIds = []
         self.sortInput = sortInput
         self.modelSchema = modelSchema
     }
 
     /// Apply the incoming `model` to the sorted array based on the mutation type. This logic accounts for duplicate
     /// events since identical events may have different sources (local and remote). When the mutation type is delete,
-    /// remove it if it exists in the array. When create/update, replace it if it exists in the array. If it does not
-    /// exist in the array, then add the model to the array in the correct position.
-    func apply(model: ModelType, mutationType: MutationEvent.MutationType) {
+    /// remove it if it exists in the array. When create/update and it has an sort order, then remove and add it back in the
+    /// correct sort order. if there is no sort order, replace it.
+    /// Return `true` if something occured (added, replaced, deleted), otherwise `false`
+    @discardableResult func apply(model: ModelType, mutationType: MutationEvent.MutationType) -> Bool {
         if mutationType == MutationEvent.MutationType.delete {
-            if let index = sortedModels.firstIndex(where: { $0.id == model.id }) {
-                sortedModels.remove(at: index)
+            return remove(model)
+        }
+
+        guard let sortInputs = sortInput else {
+            // If there is no sort order, check if it exists to replace, otherwise add it to the end
+            if modelIds.contains(model.id), let index = sortedModels.firstIndex(where: { $0.id == model.id }) {
+                sortedModels[index] = model
+            } else {
+                sortedModels.append(model)
```

Comment:
can we move 
```
sortedModels.append(model)
modelIds.insert(model.id)
``` 
in a separate function, to make sure both are happening together to keep parity?
like `remove()` function

Reply:
i'm not sure exactly what the name of the separate function you are thinking of since these two lines are only ever called once, however to keep things in parity I extracted the additional logic above these two lines into a method called `appendOrReplace(model)`, which has a similar structure to `remove` so there's some parity here now